### DISCUSSION TITLE
Ready for small projects?

### DISCUSSION THREAD

**Discussion by freedxmgxd** [carbon-language/carbon-lang#4445] [Getting started]
I am about to start a small project, using C++, Cantera, and OpenFoam, and I expect to migrate them to Carbon eventually, so I want to know if the actual releases are enough to evaluate C++ code and after trying carbon code too.

**Comment by jonmeow** [comment]
Carbon is still in early development. Right now, it can compile code, but we are still building features to make it usable. The [sieve example](https://github.com/carbon-language/carbon-lang/blob/trunk/examples/sieve.carbon) shows where things are (needing lots of boilerplate for i32 impls, for example, although zygoloid is getting close to fixing that).

The main evaluation version will be the 0.1 milestone, which is [on the roadmap for 2025](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/roadmap.md#potential-2025-goals-ship-a-working-01-language-for-evaluation). I expect it'll start being more usable over the next year, but building the full [0.1 feature list](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/milestones.md#language-features) would optimistically be late 2025.

We're happy to have the interest, just a lot of things missing still.

**Reply by tahadostifam** [reply]
I'm eager to use carbon! The idea behind it is lovely so much.

But the problem is that I don't know how i can contribute to Carbon. I know about interpreters and compilers, and also I can help with documentation and improving the official website, but I don't know from which point I have to start.

Would you give me some clues?

**Reply by jonmeow** [reply]
We mostly collaborate on [Discord](https://discord.gg/ZjVdShJDAs) -- #contributing-help and #toolchain are good spots to ask questions. On the toolchain, we keep a [tasks doc](https://docs.google.com/document/d/1O2rcDO70BamTERrKhJRLODarNCQOHQWUweNdSfLw-So/edit?resourcekey=0-gWJkSd-tlpCmk2lok1UcPg&tab=t.0#heading=h.q5lydnc75024) that we use to track ongoing work. Some tasks there are noted as smaller, although still require taking some time to understand the toolchain.

For documentation, I'm just not aware of any specific issues waiting to be addressed. We broadly continue to work on proposals, and I'm sure there are some spots where the design is out-of-date due to proposals, but I think we've mostly kept it up to date.
